File: AuthenticationServices/src/main/java/com/artauction/authenticationservices/Entities/Buyer.java
package com.artauction.authenticationservices.Entities;

import jakarta.persistence.Column;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import lombok.Getter;
import lombok.Setter;

@Entity
@DiscriminatorValue("Buyer")
@Getter
@Setter
public class Buyer extends User {
    // Attributes
    @Column(length = 50)
    private String city;

    @Column(length = 50)
    private String country;

    @Column(length = 150)
    private String address;

    @Column(length = 13)
    private String phoneNumber;

    // Relationships live in AuctionServices and ArtworkPostServices
}

File: Project/Backend/src/ArtAuction.Application/Entities/Buyer.cs
namespace ArtAuction.Application.Entities;

public class Buyer : ApplicationUser
{
    // Attributes
    public string City { get; set; } 
    public string Country { get; set; }
    public string Address { get; set; } 
    
    // Relationships
    public ICollection<WatchList> WatchLists { get; set; }
    public ICollection<PostBid> PostBids { get; set; }
    public ICollection<PostSold> PostSolds { get; set; }
}   